features/shopping_lists/src/main/kotlin/gq/kirmanak/mealient/shopping_lists/ui/details/ShoppingListScreen.kt

```diff
@@ -145,47 +147,79 @@ private fun ShoppingListScreen(
+                        val state = remember {
+                            ShoppingListItemEditorState(
+                                state = itemState,
+                                foods = loadingState.data?.foods.orEmpty(),
+                                units = loadingState.data?.units.orEmpty(),
+                            )
+                        }
+                        ShoppingListItemEditor(
+                            state = state,
+                            onEditCancelled = { onEditCancel(itemState) },
+                            onEditConfirmed = { onEditConfirm(itemState, state) },
+                        )
+                    } else {
+                        ShoppingListItem(
+                            itemState = itemState,
+                            showDivider = firstCheckedItemIndex == index,
+                            modifier = Modifier.background(MaterialTheme.colorScheme.surface),
+                            onCheckedChange = { onItemCheckedChange(itemState, it) },
+                            onDismissed = { onDeleteItem(itemState) },
+                            onEditStart = { onEditStart(itemState) },
                         )
                     }
+                } else if (itemState is ShoppingListItemState.NewItem) {
                     ShoppingListItemEditor(
-                        state = state,
-                        onEditCancelled = { onEditCancel(itemState) },
-                        onEditConfirmed = { onEditConfirm(itemState, state) },
-                    )
-                } else {
-                    ShoppingListItem(
-                        itemState = itemState,
-                        showDivider = index == firstCheckedItemIndex && index != 0,
-                        modifier = Modifier.background(MaterialTheme.colorScheme.surface),
-                        onCheckedChange = { onItemCheckedChange(itemState, it) },
-                        onDismissed = { onDeleteItem(itemState) },
-                        onEditStart = { onEditStart(itemState) },
+                        state = itemState.item,
+                        onEditCancelled = { onAddCancel(itemState) },
+                        onEditConfirmed = { onAddConfirm(itemState) }
                     )
                 }
-            } else if (itemState is ShoppingListItemState.NewItem) {
-                ShoppingListItemEditor(
-                    state = itemState.item,
-                    onEditCancelled = { onAddCancel(itemState) },
-                    onEditConfirmed = { onAddConfirm(itemState) }
-                )
             }
         }
     }
 }
 
+@Composable
+fun ShoppingListSectionHeader(labelGroup: ItemLabelGroup) {
+    // Display default label if necessary
+    val displayLabel = when (labelGroup) {
+        is ItemLabelGroup.DefaultLabel -> stringResource(id = R.string.shopping_lists_screen_default_label)
+        is ItemLabelGroup.Label -> labelGroup.name
+        else -> ""
```

Comment:
```suggestion
        is ItemLabelGroup.CheckedItems -> return
```

This has two advantages:
1. when becomes exhaustive, so if a new group is added it will have to be handled here otherwise the code won't compile
2. There is no need to check the label twice, the header is not drawn even if this function is called. So if statement on line 161 can be removed.

What do you think?

Reply:
This is implemented within [ee084b7](https://github.com/kirmanak/Mealient/pull/319/commits/ee084b763be33267a008d6f1ba918d3eefbdc9a4) :+1: